### ISSUE TITLE
Context detection issue in MA0004

### ISSUE THREAD

**Issue by Smallgiil** [meziantou/Meziantou.Analyzer#682]
### Version of the Meziantou.Analyzer NuGet package

2.0.133

### Rule Identifier

MA0004

### Target Framework

net472

### C# Language version

C# 9

### Description

Sometimes, it is legit to not specify `ConfigureAwait(false)` when awaiting. The analyzer offer an option (`MA0004.report = DetectContext`) to detect the context and remove the message when it is OK. However, the context detection is not 100% bulletproof.

For instance, in my WPF app, when I await in a view model class, the warning is raised but writting `ConfigureAwait(false)` would cause a bug.

I don't think it is possible to have a context detection that is always 100% bulletproof in 100% of the scenarii. But we can discuss it. Maybe you have good ideas.

If we agree that it's not possible, then I would simply rephrase the message:
Current: `Use Task.ConfigureAwait(false) as the current SynchronizationContext is not needed`
Proposed: `If the current SynchronizationContext is not needed, use Task.ConfigureAwait(false). Otherwise use Task.ConfigureAwait(true) or locally disable MA0004.`

What do you think?

### Reproduction Steps

Minimal code:

```c#
code to reproduce the error
```

**Comment by Smallgiil** [comment]
Thanks!
